binderhub/build.py

```diff
@@ -91,6 +91,18 @@ class BuildExecutor(LoggingConfigurable):
         config=True,
     )
 
+    push_secret_content = Unicode(
+        "",
+        help=(
+            "Content of an implementation dependent secret for pushing image to a registry. "
+            "For example, if push tokens are temporary this can be used to pass the token "
+            "as an environment variable CONTAINER_ENGINE_REGISTRY_CREDENTIALS to "
+            "repo2docker."
+            "If provided this will be used instead of push_secret."
+        ),
+        config=True,
+    )
+
```

Comment:
Overall, the `push_secret` config has been challenging for me to understand well. Here are some notes on what I think can help.

- Its not obvious that its a `push_secret` references a name (k8s Secret resource for example) as compared to content to put inside one.
- Its not obvious that its going to be created or needs to be created manually
- Its not obvious what kind of content to put in the referenced secret created manually

Thinking about what I'd appreciate from docs to understand this good enough, I'd say something like the following, where its not obvious whats in scope and not for this PR:

1. I think "implementation dependant" could be written more verbosively to clarify that: its dependant on the BuildExecutor implementation, and no base implemention exist.
2. `push_secret` and `push_secret_content`'s help strings are both are updated to reflect that there are separate ways of providing container registry credentials to the builder, where `push_secret` references something existing by name, and `push_secret_content` is content.
3. `push_secret` and `push_secret_content`'s help strings are both are updated on how to know what kind of resource to reference and what content to provide. This may be implementation dependent and not something BuildExecutor can know, but I think we can declare here where to find that kind of details.

Reply:
I'll update the docs. AFAIK it should be fine to duplicate the traitlet in the subclass to have more specialist documentation?

If it's easier (I'm not sure whether it is), we could switch `push_secret` to be of type `(Unicode, Dict)`. where Dict would for example take the form `{content: <secret>}` or `{secret_name: <k8s-secret-name>}`. `Unicode` would support the current default behaviour.